framework/src/engine/consensus/consensus.ts

```diff
@@ -308,7 +322,9 @@ export class Consensus {
 		this.events.emit(CONSENSUS_EVENT_NETWORK_BLOCK_NEW, { block });
 
 		try {
+			const endExecuteMetrics = this._metrics.blockExecution.startTimer();
 			await this._execute(block, peerId);
+			endExecuteMetrics();
```

Comment:
I hope its the correct usage of endExecuteMetrics, does this._metrics.blockExecution.startTimer() returns a functions ?

Reply:
yes, the startTimer returns function to end the timer. (also, even if the end is not called, it wont leak anything)